### PR TITLE
[dtensor] move DTensor to public namespace

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133113
* #133306
* #133305

Moving DTensor to be in the public namespace, to formally add the
documentation page that includes all the public APIs. This includes:

* many path renames and path import fixes
* a dedicated doc page without too much content yet (adding in the next
  PRs)
* To preserve the BC for users still using the `torch.distributed._tensor`,
  I added a shim script to redirect old path calls to the new module

The BC preserving is evidented by the fact that all DTensor tests are still
working without changing the public imports. So it's safe to land the
changes

cc @XilunWu @H-Huang @awgu @kwen2501 @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec @LucasLLC @MeetVadakkanchery @mhorowitz @pradeepfn

### CODE REVIEW

**Review by XilunWu — APPROVED** [review]
the distributed part lgtm

File: test/distributed/_tensor/test_common_rules.py [hunk 1]
```diff
@@ -3,9 +3,9 @@
 
 import torch
 from torch.distributed._tensor import DeviceMesh
-from torch.distributed._tensor._op_schema import OpSchema
-from torch.distributed._tensor.ops._common_rules import einop_rule, pointwise_rule
 from torch.distributed._tensor.placement_types import DTensorSpec, TensorMeta
```
[the remark this answers is not in the document]
Comment (wanchaol) [reply test/distributed/_tensor/test_common_rules.py:6]: It's because we want to make sure we are still backward compatible to old imports

**Comment by wanchaol** [comment]
@wanchaol has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61458680).

**Pull request closed without merging**